src/trusted/service_runtime/sel_mem.c

```diff
@@ -579,6 +581,75 @@ int NaClVmmapCheckExistingMapping(struct NaClVmmap  *self,
   return 0;
 }
 
+int NaClVmmapCheckAddrMapping(struct NaClVmmap  *self,
+                              uintptr_t         page_num,
+                              size_t            npages,
+                              int               prot) {
+  size_t      i;
+  uintptr_t   region_end_page = page_num + npages;
+
+  NaClLog(2,
+          ("NaClVmmapCheckExistingMapping(0x%08"NACL_PRIxPTR", 0x%"NACL_PRIxPTR
+           ", 0x%"NACL_PRIxS", 0x%x)\n"),
+          (uintptr_t) self, page_num, npages, prot);
+
+  if (0 == self->nvalid) {
+    return 0;
+  }
+
+  // Caching entries to improve I/O speed
+
+  if (self->cached_entry) {
+    struct NaClVmmapEntry   *ent = self->cached_entry;
+    uintptr_t               ent_end_page = ent->page_num + ent->npages;
+    int                     flags = ent->prot;
+   
+    //If the page does not not have PROT_NONE, force the PROT_READ flag
+    //This behavior is unspeicified by POSIX, but Linux acts in this way
+    if((flags & (NACL_ABI_PROT_EXEC | NACL_ABI_PROT_READ | 
+                NACL_ABI_PROT_WRITE)) != PROT_NONE){
+        flags |= PROT_READ;
+    }
+
+    if (ent->page_num <= page_num && region_end_page <= ent_end_page) {
```

Comment:
I think(?) the ent_end_page page is one greater than what we actually want, so do a < there? I'm really not sure of this, so confirm the behavior here

Reply:
It's not, this is the same as the check in the original function.